pkg/controller/certificaterequest/certificaterequest_controller.go

```diff
@@ -24,132 +24,153 @@ import (
 	"math/rand"
 
 	"github.com/go-logr/logr"
-	apiutil "github.com/jetstack/cert-manager/pkg/api/util"
+	cmutil "github.com/jetstack/cert-manager/pkg/api/util"
 	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
 	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
 	api "github.com/jetstack/google-cas-issuer/api/v1alpha1"
 	"github.com/jetstack/google-cas-issuer/pkg/cas"
 	issuerctrl "github.com/jetstack/google-cas-issuer/pkg/controller/issuer"
-	core "k8s.io/api/core/v1"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/client-go/tools/record"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	casapi "github.com/jetstack/google-cas-issuer/api/v1alpha1"
 )
 
 // CertificateRequestReconciler reconciles CSRs
 type CertificateRequestReconciler struct {
 	client.Client
-	Log      logr.Logger
-	Recorder record.EventRecorder
+	Log logr.Logger
 }
 
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
 // +kubebuilder:rbac:groups=cert-manager.io,resources=certificaterequests,verbs=get;list;watch;update
```

Comment:
Does the controller need rw permissions on CertificateRequest.{ObjectMeta, Spec} ? if not then might be worth removing this `update` permission in another PR.